datalad/core/distributed/clone.py

```diff
@@ -581,21 +581,141 @@ def postclonecfg_ria(ds, props):
-                   if s.get('annex-externaltype', None) == 'ora'
-    ]
-    if not ria_remotes:
-        lgr.debug("Found no RIA special remote")
-    elif len(ria_remotes) == 1:
-        yield from ds.siblings('configure',
-                               name='origin',
-                               publish_depends=ria_remotes[0]['name'],
-                               result_filter=None,
-                               result_renderer='disabled')
-    else:
-        lgr.warning("Found multiple RIA remotes. Couldn't decide which "
-                    "publishing to origin should depend on: %s",
-                    [r['name'] for r in ria_remotes])
+    ora_remotes = [s for s in ds.siblings('query', result_renderer='disabled')
+                   if s.get('annex-externaltype', None) == 'ora']
+    if not ora_remotes and any(
+            r.get('externaltype', None) == 'ora'
+            for r in ds.repo.get_special_remotes().values()):
+        # no ORA remote autoenabled, but configuration know about at least one.
+        # Let's check origin's config for datalad.ora-remote.uuid as stored by
+        # create-sibling-ria and enable try enabling that one.
+        lgr.debug("Found no autoenabled ORA special remote. Trying to look it "
+                  "up in source config ...")
+
+        # First figure whether we cloned via SSH, HTTP or local path and then
+        # get that config file the same way:
+        config_content = None
+        scheme = props['giturl'].split(':')[0]
+        if scheme == 'http':
+
+            # Note: Following outcommented code was proof how to check via HTTP.
+            #       However, for now we only want to annexenable origin in case
+            #       of HTTP, not the special remote. Hence it's uuid isn't
+            #       actually required. (see gh-4410)
+
+            # import requests
+            # try:
+            #     response = requests.get("{}{}config".format(
+            #         props['giturl'],
+            #         '/' if not props['giturl'].endswith('/') else '')
+            #     )
+            #     config_content = response.text
+            # except requests.RequestException as e:
+            #     lgr.debug("Failed to get config file from source:\n%s",
+            #               exc_str(e))
+
+            lgr.debug("No ORA reconfiguration for HTTP implemented yet")
+
+        elif scheme == 'ssh':
+
+            # TODO: switch the following to proper command abstraction:
+            from datalad.distributed.ora_remote import SSHRemoteIO, RIARemoteError
+            # SSHRemoteIO ignores the path part ATM. No remote CWD! (To be
+            # changed with command abstractions). So we need to get that part to
+            # have a valid path to origin's config file:
+            from datalad.support.network import URL
+            from datalad.utils import PurePosixPath
+            cfg_path = PurePosixPath(URL(props['giturl']).path) / 'config'
```

Comment:
it might be good to program a bit more defensively here and try/except for KeyError in case `giturl` was not defined, and spit out a specific meaningful exception describing what is missing `giturl` which is required

Reply:
What would be a useful description here? It's the URL we just cloned from. If it's not in `props` we have a bug in `clone_dataset`. I wouldn't know a better message for a user. They can't do anything about it. And all we need to know to fix it is that KeyError.